packages/backend/src/apps/formsg/triggers/new-submission/get-data-out-metadata.ts

```diff
@@ -51,42 +51,71 @@ function buildAnswerMetadatum(fieldData: IJSONObject): IDataOutMetadatum {
   return answer
 }
 
-function buildAnswerArrayMetadatum(
+function isAnswerArrayValid(fieldData: IJSONObject): boolean {
+  if (!fieldData.answerArray) {
+    return false
+  }
+  // strict check for only table and checkbox variables
+  return fieldData.fieldType === 'table' || fieldData.fieldType === 'checkbox'
+}
+
+function buildAnswerArrayForCheckbox(
   fieldData: IJSONObject,
 ): IDataOutMetadatum[] {
-  // there should only be checkbox and table fieldtypes that contain answer array
-  const answerArray = [] as IDataOutMetadata[]
+  const answerArray = [] as IDataOutMetadatum[]
+  const array = fieldData.answerArray as IJSONArray
+  for (let i = 0; i < array.length; i++) {
+    answerArray.push({
+      type: 'text',
+      label: fieldData.order
+        ? `Response ${fieldData.order}, Selected Option ${i + 1}`
+        : null,
+      order: fieldData.order ? (fieldData.order as number) : null,
+    })
+  }
+  return answerArray
+}
+
+function buildAnswerArrayForTable(
+  fieldData: IJSONObject,
+): IDataOutMetadatum[][] {
+  const answerArray = [] as IDataOutMetadatum[][]
   const array = fieldData.answerArray as IJSONArray
   for (let i = 0; i < array.length; i++) {
-    // check for table fieldType to do more processing
     const option = array[i]
-    if (fieldData.fieldType === 'table') {
-      const nestedAnswerArray = [] as IDataOutMetadatum[]
-      const optionArray = option as IJSONArray
-      for (let j = 0; j < optionArray.length; j++) {
-        nestedAnswerArray.push({
-          type: 'text',
-          label: fieldData.order
-            ? `Response ${fieldData.order}, Row ${i + 1} Column ${j + 1}`
-            : null,
-          order: fieldData.order ? (fieldData.order as number) : null,
-        })
-      }
-      answerArray.push(nestedAnswerArray)
-    } else {
-      // checkbox fieldtype processing
-      answerArray.push({
+    const nestedAnswerArray = [] as IDataOutMetadatum[]
+    const optionArray = option as IJSONArray
+    for (let j = 0; j < optionArray.length; j++) {
+      nestedAnswerArray.push({
         type: 'text',
         label: fieldData.order
-          ? `Response ${fieldData.order}, Selected Option ${i + 1}`
+          ? `Response ${fieldData.order}, Row ${i + 1} Column ${j + 1}`
           : null,
         order: fieldData.order ? (fieldData.order as number) : null,
       })
     }
+    answerArray.push(nestedAnswerArray)
   }
   return answerArray
 }
 
+function buildAnswerArrayMetadatum(
+  fieldData: IJSONObject,
+): IDataOutMetadatum[] | IDataOutMetadatum[][] {
+  // there should only be checkbox and table fieldtypes that contain answer array
+  switch (fieldData.fieldType) {
+    case 'checkbox':
+      return buildAnswerArrayForCheckbox(fieldData)
+    case 'table':
+      return buildAnswerArrayForTable(fieldData)
+    default:
+      console.log(
```

Comment:
Don't console.log in production code pls. Use `logger`

Reply:
Also make sure to include structured data so that we can grep and filter for it in datadog.

Example: https://github.com/opengovsg/plumber/blob/eef3a772010dbd3305e96c3a4a7b93683195f48d/packages/backend/src/controllers/webhooks/handler.ts#L62

Reply:
my bad, ok will do so thanks!